src/assets/sql/stored-procedure/countAppointmentWithUserIdAndServiceIdAndPatientId.sql

```diff
@@ -0,0 +1,69 @@
+-------------------------------------------------------------------------- STORED PROCEDURE SCRIPT----------------------------------------------------------------
+DELIMITER $$
+DROP PROCEDURE IF EXISTS `countAppointmentsByCondition` $$
+CREATE DEFINER=`root`@`localhost` PROCEDURE `countAppointmentsByCondition`(IN `idUserIntegrated` VARCHAR(200), 
+                                                                           IN `idServiceIntegrated` VARCHAR(200), 
+                                                                           IN `idPatientIntegrated` VARCHAR(200), 
+                                                                           IN `startDate` VARCHAR(200), 
+                                                                           IN `endDate` VARCHAR(200))
+BEGIN
+	Declare startOffset Integer;
+    #-- Prepare pre-condition before running main query
+    IF idServiceIntegrated <> '' THEN
+        SET idServiceIntegrated = (SELECT ea_services.id from ea_services WHERE ea_services.id_integrated = idServiceIntegrated);
```

Comment:
why do you need to transform to id from id_service_integrated?
can you fix the finalQuery to make joining tables then put id_service_integrated to the WHERE statement directly
this is to reduce this kind of query, ?

Reply:
I already updated the query statement